app/src/main/java/com/github/sdp_begreen/begreen/fragments/MapFragment.kt

```diff
@@ -1,70 +1,128 @@
+        map.uiSettings.isZoomControlsEnabled = true
+
+        checkUserLocationPermissions()
+    }
+
+    private val requestPermissionLauncher = registerForActivityResult(ActivityResultContracts.RequestPermission()) { isGranted ->
+
+        if (isGranted) {
+            displayUserLocation()
         }
     }
 
     override fun onCreateView(
-        inflater: LayoutInflater, container: ViewGroup?,
+        inflater: LayoutInflater,
+        container: ViewGroup?,
         savedInstanceState: Bundle?
     ): View? {
-        // Inflate the layout for this fragment
-        val view: View? = inflater.inflate(R.layout.fragment_map, container, false)
-        val tView: TextView? = view?.findViewById(R.id.mapFragmentTextView)
+        return inflater.inflate(R.layout.fragment_map, container, false)
+    }
 
-        tView?.text = getString(
-            R.string.map_fragment_text,
-            param1?.let { " $it" }.orEmpty(),
-            param2?.let { ", $it" }.orEmpty()
-        )
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        val mapFragment = childFragmentManager.findFragmentById(R.id.mapFragment) as SupportMapFragment?
+        mapFragment?.getMapAsync(mapReadyCallback)
 
-        return view
+        // Get the localisation of the user
+        fusedLocationClient = LocationServices.getFusedLocationProviderClient(requireActivity())
     }
 
-    companion object {
-        /**
-         * Use this factory method to create a new instance of
-         * this fragment using the provided parameters.
-         *
-         * @param param1 Parameter 1.
-         * @param param2 Parameter 2.
-         * @return A new instance of fragment Home.
-         */
-        // TODO: Rename and change types and number of parameters
-        @JvmStatic
-        fun newInstance(param1: String, param2: String) =
-            MapFragment().apply {
-                arguments = Bundle().apply {
-                    putString(ARG_PARAM1, param1)
-                    putString(ARG_PARAM2, param2)
-                }
+    /**
+     * Displays the user current location. Asks for permissions if needed.
+     */
+    private fun checkUserLocationPermissions() {
+
+        if (ContextCompat.checkSelfPermission(
+                requireContext(),
+                Manifest.permission.ACCESS_FINE_LOCATION
+            ) != PackageManager.PERMISSION_GRANTED) {
+
+            requestPermissionLauncher.launch(Manifest.permission.ACCESS_FINE_LOCATION)
+
+        } else {
+
+            displayUserLocation()
+        }
+    }
+
+    /**
+     * Displays the user current location assuming that location permissions are granted
+     */
+    private fun displayUserLocation() {
+
+        if (ContextCompat.checkSelfPermission(
+                requireContext(),
+                Manifest.permission.ACCESS_FINE_LOCATION
+            ) == PackageManager.PERMISSION_GRANTED) {
```

Comment:
I think that this test is a bit redundant, as the only way to call the method `displayUserLocation` is when the permission was either already granted, or has just been granted.

I think we could safely remove it. What do you think ?

Reply:
At first I was not testing the permissions again but I had a warning saying that I had to add permission checks.
So I decided to check it again. But I've just found a way to ignore the warning with the following annotation :

`@SuppressLint("MissingPermission")`

Therefore I can remove the redundant check. Glad we got the same sighting !